engine/runtime-parser/src/test/java/org/enso/compiler/core/IrPersistanceTest.java

```diff
@@ -68,7 +69,37 @@ public void identifiedLocationNoUUID() throws Exception {
   }
 
   @Test
-  public void idHolderNoUUID() throws Exception {
+  public void refHolderWithUUID() throws Exception {
+    var id = UUID.randomUUID();
+    var il = new RefHolder(id);
+    var in = serde(RefHolder.class, il, -1, null);
```

Comment:
This stores an instance of `RefHolder` and reads another instace of `RefHolder` back from the array. Only then when one `in.id().get(UUID.class)` the instance of `UUID` gets deserialized.